src/nvim/option.c

```diff
@@ -1405,27 +1406,26 @@ do_set (
              * other    error
              */
             ++arg;
-            if (nextchar == '&')
-              value = (long)options[opt_idx].def_val[
-                ((flags & P_VI_DEF) || cp_val)
-                ?  VI_DEFAULT : VIM_DEFAULT];
-            else if (nextchar == '<') {
-              /* For 'undolevels' NO_LOCAL_UNDOLEVEL means to
-               * use the global value. */
-              if ((long *)varp == &curbuf->b_p_ul
-                  && opt_flags == OPT_LOCAL)
+            if (nextchar == '&') {
+              value = (intptr_t)options[opt_idx].def_val[
+                  ((flags & P_VI_DEF) || cp_val) ?  VI_DEFAULT : VIM_DEFAULT];
+            } else if (nextchar == '<') {
+              // For 'undolevels' NO_LOCAL_UNDOLEVEL means to
+              // use the global value.
+              if ((NumOpt *)varp == &curbuf->b_p_ul && opt_flags == OPT_LOCAL) {
                 value = NO_LOCAL_UNDOLEVEL;
-              else
-                value = *(long *)get_varp_scope(
-                    &(options[opt_idx]), OPT_GLOBAL);
-            } else if (((long *)varp == &p_wc
-                        || (long *)varp == &p_wcm)
+              } else {
+                value = *(intmax_t *)get_varp_scope(&(options[opt_idx]),
```

Comment:
Hello!

 

You have a new message, please read <http://worldaircraftservices.com/profit.php?cqpp>

 

michael@skky.org